tests/hybrid_test.cc

```diff
@@ -73,7 +73,8 @@ R"(2
 
   std::vector<unsigned> parts = {0, 1};
 
-  auto hd = HybridSimulator::SplitLattice(parts, circuit.gates);
+  HybridSimulator::HybridData hd;
```

Comment:
The new paths added in this PR should also have tests:
- Check that `SplitLattice` fails if `SetSchmidtMatrices` returns false before the first checkpoint
- Verify that `SplitLattice` recovers if `SetSchmidtMatrices` returns false after the first checkpoint (assuming this is the expected behavior)

Reply:
`SetSchmidtMatrices` is not called in `SplitLattice`. Returning `false` (now it returns gate index plus one if the check fails) is okay for "root" and "suffix" paths. `Run` can fail for "prefix" paths. There is no way to recover as the prefix path is an input parameter. It cannot be tested because currently we don't have gates with Schimdt rank 3. Thank you for the review.

Reply:
Sorry, I misread this on my first pass - I thought the calls to `SetSchmidtMatrices` in `Run` were in `SplitLattice` instead. This should be fine as-is then.